rich/containers.py

```diff
@@ -137,7 +135,7 @@ def justify(
                 line.pad_right(width - cell_len(line.plain))
         elif justify == "right":
             for line in self._lines:
-                line.rstrip()
+                line[:-1].rstrip() if line[-1] == Text(" ") else line.rstrip()
```

Comment:
Since the markup is irrelevent for the condition, it is faster to do `line.plain.endswith(" ")`